rpc/transport/connection.go

```diff
@@ -1,18 +1,32 @@
 package transport
 
-import "github.com/statechannels/go-nitro/rpc/serde"
+type ConnectionType string
 
-type Connection interface {
-	// Request sends data for a topic and returns the response data or an error
-	Request(serde.RequestMethod, []byte) ([]byte, error)
-	// Respond listens for requests for a topic and calls the handler function when a request is received
-	Respond(serde.RequestMethod, func([]byte) []byte) error
+const (
+	Nats ConnectionType = "nats"
+	Ws   ConnectionType = "ws"
+)
 
-	// Notify sends data for a topic without expecting a response
-	Notify(serde.NotificationMethod, []byte) error
+// Requester is a connection that can send requests and subscribe to notifications
+type Requester interface {
+	// Close closes the connection
+	Close()
+
+	// Request sends data for a topic and returns the response data or an error
+	Request(uint64, []byte) ([]byte, error)
 	// Subscribe listens for notifications for a topic and does not respond to notifications
-	Subscribe(serde.NotificationMethod, func([]byte)) error
+	Subscribe() (<-chan []byte, error)
+}
 
-	// Close shuts down the connection
+// Responder is a connection that can respond to requests and send notifications
+type Responder interface {
+	// Close closes the connection
 	Close()
+	// Url returns the url that the responder is listening on
+	Url() string
```

Comment:
Not sure I fully understand the question unfortunately. `transport.Requester` and `transport.Responder` are interfaces that are implemented by transports like nats or websockets. `RpcClient` and `RpcServer` are implementations of a client and server that can use any transport for communication.

Would you mind elaborating?

Reply:
What if we had 
- `NATSRPCServer`
- `WebsocketsRPCServer`
- etc

implementations, instead of a single
- `RPC Server`

implementation with an injected dependency conforming to a `Responder` interface? I guess the current solution will be better if the list of transports we might use gets longer. This is low priority / don't feel strongly... just want to explore the solution space a bit.

Reply:
Interesting idea! It seems like the transport specific servers would need to share the helper code like [this](https://github.com/statechannels/go-nitro/blob/fc11b3d61b6e60631688b2533a0cf34fe92e90c1/rpc/server.go#L47). Do you invision creating a base server that would contain code like this?

Reply:
Good point. I hadn't really thought about the work the servers do tying together the serde code with the transport code. Let's continue on with the current pattern.